.github/workflows/test-metallb-helm-chart.yml

```diff
@@ -0,0 +1,165 @@
+            echo "${file}"
+          done
+          exit 1  # Fail the workflow
+        fi
+        
+        mkdir helm-charts
+        tar xvzf "${archive}" -C helm-charts/
+
+    - name: Install K3s
+      run: |
+        curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="server --cluster-init --write-kubeconfig-mode=644 --disable=servicelb --disable=traefik" K3S_TOKEN=foobar sh -
+
+    - name: Install Helm
+      run: |
+        curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
+        chmod +x get_helm.sh
+        ./get_helm.sh
+        
+    - name: Deploy metallb Helm chart
+      run: |
+        export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+        helm install metallb helm-charts/metallb
+        
+    - name: Wait for all metallb pods to become ready
+      run: |
+        export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+        
+        # Set a timeout of 5 minutes
+        timeout=$((SECONDS + 300))
+        
+        while [ $SECONDS -lt $timeout ]; do
+            # Run the kubectl command to get pod information
+            kubectl_output=$(kubectl -n default get po | tail -n +2)
+        
+            # Flag to track whether all pods are ready
+            all_pods_ready=true
+        
+            # Iterate over each line in the kubectl output
+            while IFS= read -r line; do
+                # Extract the pod name and the readiness status
+                pod_name=$(echo "$line" | awk '{print $1}')
+                readiness_status=$(echo "$line" | awk '{print $2}')
+        
+                # Extract the desired and running replicas from the readiness status
+                desired_replicas=$(echo "$readiness_status" | awk -F'/' '{print $1}')
+                running_replicas=$(echo "$readiness_status" | awk -F'/' '{print $2}')
+        
+                # Check if the digit before / is the same as the one after /
+                if [ "$desired_replicas" -eq "$running_replicas" ]; then
+                    echo "$pod_name is ready"
+                else
+                    echo "$pod_name is not ready"
+                    all_pods_ready=false
+                fi
+            done <<< "$kubectl_output"
+        
+            # Check if all pods are ready
+            if [ "$all_pods_ready" = true ]; then
+                echo "All pods are ready"
+                exit 0
+            fi
+        
+            # Wait for a moment before checking again
+            sleep 5
+        done
+        
+        # If the loop completes, it means the timeout occurred
+        echo "Timeout: Not all pods are ready in 5 minutes."
+        exit 1
+
+    - name: Define IPAddressPool
+      run: |
+        export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+
+        cat <<-EOF | kubectl apply -f -
+        apiVersion: metallb.io/v1beta1
+        kind: IPAddressPool
+        metadata:
+          name: ip-pool
+        spec:
+          addresses:
+          - 1.1.1.1/32
```

Comment:
This is an actual resolvable IP - I can't see how we can reserve a real IP (unless we use a self-hosted runner), perhaps this should be an IP from the `TEST-NET-1` range to make it clear it's only for testing?

Reply:
Okay, what about `192.0.2.1`?